### ISSUE TITLE
MikTeX installation fails with error “Not a file name database file (wrong signature)”

### ISSUE THREAD

**Issue by NikitaSperling** [MiKTeX/miktex#95]
I am running MikTeX 2.9 on Windows 10 and updated my MikTeX packages yesterday as an admin and afterwards as a normal user. During the last update, the MikTeX console and all other programs, that were running, stopped responding and nothing worked anymore so I had to push the power button. After restarting I could not compile any LaTeX document anymore. When I tried to open the MikTeX console (no matter if setting, update or package mode or if admin or user mode) I got the error `Not a file name database file (wrong signature)`.

So I wanted to deinstall and reinstall the program but even on deinstalling (whether about the uninstall.exe or the control panel in Windows) I got the same error and could not even deinstall it anymore. So I deleted the MikTeX folder from the programs directory and cleaned my computer and registry with CCleaner.

Then I tried to newly install MikTeX. At the first glance, this seemed to work but just at the end of the installation MikTeX again fails with this error:

    Not a file name database file (wrong signature).
    Info: path="C:\ProgramData\MiKTeX\2.9\miktex/data/le\1756f6ade37d5f870f23c05a7f81e257.fndb"

I tried to install for all users (admin mode) and for one user several times including several restarts of the computer, but always get the same error. I turned off my antivirus program and firewall and selected "ask me first" when asked to install packages on the fly. That didn't help either, still the same error. I'd been trying for hours but didn't succeed. Windows made an update the night before. Maybe it has something to do with that?

As far as I can see, the installation was almost finished. There are all folders and .exe-files present in the program directories as I had before deinstallation. I don't understand why it doesn't work and what the error means. These are the last lines of the last .log-file:

    initexmf "--common-install=C:\Program Files\MiKTeX 2.9" --set-config-value=[Core]SharedSetup=1 --admin "--log-file=C:\Program Files\MiKTeX 2.9\miktex/config\uninst.log" --disable-installer --verbose:
    Not a file name database file (wrong signature).
    Info: path="C:\ProgramData\MiKTeX\2.9\miktex/data/le\1756f6ade37d5f870f23c05a7f81e257.fndb"
    Source: Libraries\MiKTeX\Core\Fndb\FileNameDatabase.cpp
    Line: 99

    Sorry, but "MiKTeX Configuration Utility" did not succeed.

    You may want to visit the MiKTeX project page, if you need help.

    An error occurred:
      source file: Libraries\MiKTeX\Core\Process\Process.cpp
      source line: 166
      message: The executed process did not succeed.
      info: fileName="C:\Program Files\MiKTeX 2.9\miktex/bin/x64\initexmf.exe", exitCode="1"

After asking for help at [TeX Stack Exchange](https://tex.stackexchange.com/questions/420619/miktex-installation-fails-with-error-not-a-file-name-database-file-wrong-signa?noredirect=1#comment1053068_420619), I installed MikTeX portable. With this workaround I succeeded to compile my LaTeX documents again. However, I still wonder what the problem with the basic MikTeX installation is.

**Comment by edocevoli** [comment]
Seems that the file name database file is corrupted. I would recommend that you remove it and try again.

**Comment by edocevoli** [comment]
FYI: file name database files start with four ASCII characters: `F`, `N`, `D`, `B`. The error message (`wrong signature`) indicates that this is not the case. I would recommend that you check your system for malware and other issues (e.g., bad hard disk blocks).

**Comment by NikitaSperling** [comment]
Thank you, @edocevoli ! I deleted and deinstalled MikTeX completely yesterday and could not find any files left. However, I now started a search for fndb on my system and it in fact found five or six fndb-files in directories that did not exist any more. Since I got an error message (file could not be found) when I tried to delete them, I again ran CCleaner and deleted all temporary files and cleaned the registry, where also some MikTeX entries were left. After that I ran the basic mikTeX installer, and now it installed without any error message.
